### DISCUSSION TITLE
How to make mini.completion work with nvim runtime completions

### DISCUSSION THREAD

**Discussion by craigmac** [nvim-mini/mini.nvim#248] [Q&A] [answered]
Right now when doing `vim.api.nvim` in a lua buffer I don't automatically get results from the nvim api, but if I do `vim.api.nvim<C-x><C-o>` to trigger built-in (in a pretty recent nvim commit `b:omnifunc` is set to `v:lua.vim.lsp.omnifunc` if it isn't set).

If I were to explicitly do `setlocal omnifunc=..` like described in the mini docs (inside `on_attach()` lsp callback), then would I still get the 2-stage completion feature?

Here's my lua-language-server settings (using neovim/nvim-lspconfig):

```lua
require('lspconfig').lua_ls.setup({
  settings = {
    Lua = {
      runtime = {
        version = 'LuaJIT',
      },
      diagnostics = {
        globals = { 'vim' },
      },
      workspace = {
        library = vim.api.nvim_get_runtime_file('', true),
        checkThirdParty = false,
      },
      telemetry = {
        enable = false,
      },
    },
  },
})
```

**Comment by echasnovski** [comment, marked as answer]
First of all, thanks for `library = vim.api.nvim_get_runtime_file('', true),` line. I didn't update my Lua Lsp setup in a while, mostly because it works and I tend to read Neovim help docs anyway. But replacing my previous `workspace.library` with this one made it better. Maybe even a bit too much, as it suggests from plugin codebases, so we'll see. 

Second, I am not sure what to tell you. I've just reproduced `vim.api.nvim` autocompletion with your 'lua_ls' setup and plain `require('mini.completion').setup()`. Let's try to be on the same page:

- Do you suggestions show if you press `<Ctrl> + <Space>` (default mapping for triggering two-stage completion)?
- What does executing `:echo &completefunc` and `:echo &omnifunc` show?

**Reply by craigmac** [reply]
I should have produced a minimal repro first, it's something in my runtime messing with it because I stripped it back and see it _is_ working as intended, thanks!

**Comment by craigmac** [comment]
Through bisect I've narrowed it down to this bit of code being uncommented in my init.lua, which causes the issue:

```lua
local MyAugroup = vim.api.nvim_create_augroup('MyAugroup', {})
vim.api.nvim_create_autocmd('BufEnter', {
     group = MyAugroup,
     command = 'if &bt ==# "terminal" | norm! a | endif',
     desc = 'Automatically enter terminal-mode when entering a terminal buffer'
})
```

@echasnovski I'm not sure how this is not compatible with the mini.completion BufEnter event settings. Do you see how setting this autocmd would mess it up? Is there a hidden terminal running or something, or maybe it's a load order thing with this autocmd?

**Reply by echasnovski** [reply]
No, can't think of anything why this should be incompatible with 'mini.completion'.

Is there a reason you don't use `autocmd TermOpen * startinsert`?

**Reply by craigmac** [reply]
Right. This is my error in transferring over some older vim script stuff, thanks for the hint!
